doc/source/dev/contributor/debugging_linalg_issues.rst

```diff
@@ -477,4 +477,63 @@ file to automate this and avoid the manual paths:
    machine, it's easy to accidentally pick up the wrong one during the build
    (remember: ``-lopenblas`` only says "link against *some*
    ``libopenblas.so``). If you're not sure, use ``ldd`` on the test executable
-   you built to inspect which shared library it's linked again.
+   you built to inspect which shared library it's linked against.
+
+
+Debugging linalg issues with ``gdb``
+------------------------------------
+
+When debugging linalg issues, it is sometimes useful to step through both Python
+and C code. You can use ``pdb`` for the former, and ``gdb`` for the latter.
+
+First, prepare a small python reproducer, with a breakpoint. For example::
+
+    $ cat chol.py
+    import numpy as np
+    from scipy import linalg
+    n = 40
+    np.random.seed(1234)
+    x = np.random.uniform(size=n)
+    a = x[:, None] @ x[None, :] + np.identity(n)
+
+    breakpoint()      # note a breakpoint
+    linalg.cholesky(a)
+
+Then, you will need to run it under the ``gdb`` and add a C-level breakpoint at
+the LAPACK function. This way, you execution will first stop twice: first on the
+Python breakpoint and then on the C breakpoint, and you will be able to step
+through and inspect values of both Python and C variables.
+
+To find out the LAPACK name, read the python source of the SciPy function and
+use ``nm`` on the ``.so`` library to find out the exact name.
+For the Cholesky factorization above, the LAPACK function is ``?potrf``, and the
+C name on Ubuntu linux is ``dpotrf_`` (it may be spelled with or without the
+trailing underscore, in uppper case or lower case, depending on the system).
+
+Here is an example ``gdb`` session::
+
+    $ gdb --args python chol.py
```

Comment:
Is it mandatory for the `python` executable to have been built with debug symbols?

I thought that was important at some point, maybe not anymore. IIRC that requirement sometimes annoyed me when working on a system that wasn't my own, since installing a new CPython build can be annoying.

Reply:
Apparently not anymore? I tried locally with whatever cpython binary comes from conda-forge. This are some annoyances with some variables reported as `<optimized out>` for the default built flags, but it's usable even with that.

Reply:
Yeah this requirement was removed around CPython 3.9 or so